cli/src/trades.rs

```diff
@@ -0,0 +1,76 @@
+use stellar_client::{endpoint::trades, error::Result, sync::{self, Client}};
+use stellar_resources::AssetIdentifier;
+use clap::ArgMatches;
+use super::{cursor, ordering, pager::Pager};
+
+pub fn all(client: &Client, matches: &ArgMatches) -> Result<()> {
+    let pager = Pager::from_arg(&matches);
+
+    let endpoint = {
+        let mut endpoint = trades::All::default();
+
+        if let Some(offer_id) = matches.value_of("offer_id") {
+            let offer_id = offer_id.parse::<u32>().unwrap();
+            endpoint = endpoint.with_offer_id(offer_id);
+        };
+        if let (Some(base_asset_type), Some(counter_asset_type)) = (
+            matches.value_of("base_asset_type"),
+            matches.value_of("counter_asset_type"),
+        ) {
+            let base_asset_code = matches
+                .value_of("base_asset_code")
+                .map(|code| code.to_string());
+            let base_asset_issuer = matches
+                .value_of("base_asset_issuer")
+                .map(|issuer| issuer.to_string());
+            let counter_asset_code = matches
+                .value_of("counter_asset_code")
+                .map(|code| code.to_string());
+            let counter_asset_issuer = matches
+                .value_of("counter_asset_issuer")
+                .map(|issuer| issuer.to_string());
+            let base_asset =
+                match AssetIdentifier::new(base_asset_type, base_asset_code, base_asset_issuer) {
+                    Ok(asset) => asset,
+                    Err(error) => panic!("There was a with the supplied base_asset: {:?}", error),
```

Comment:
There's no reason to panic here. You should find a way to map this error to the returned Result. Right now we are piggy backing on the errors from the client library. Perhaps we can find a way to wrap that into another error type and allow the errors to flow back so that the program exits in a controlled manner.

Reply:
@correlator Pardon my interruption into this convo/pr, but I just came across ?'s (the question mark) in a random tangent the other day. Would using a ? here allow the errors to flow back up if this function returned a Result type? Or am I way off talking about something unrelated (very likely)?

Reply:
You are absolutely right.  A `?` would allow us to short circuit and return the error and the final implementation of this will likely utilize that operation.  Right now I'm trying to decide which error struct is most appropriate to return.  The CLI currently utilizes errors from the `stellar_client` crate this error doesn't feel like it belongs in `stellar_client`.   I was admittedly cutting corners to panic rather than creating the appropriate error to return.  Going to be digging into this tonight.

Reply:
Thanks for the response. I'll admit, I copy-pasta a bit much, usually just as a starting point and I noticed similar code with a panic that I started to use -- after testing it a bit, the panic felt like an abrupt end. From my understanding, the only reason ever to panic is to deal with non-code errors, i.e. unpredictable runtime events. Coming from Javascript and 7 days worth of GO so panic and real error handing in general is pretty new.

Reply:
@robertDurst @correlator Yup, it will work very well if the type implements the `From` trait.